into/backends/aws.py

```diff
@@ -0,0 +1,248 @@
+from __future__ import print_function, division, absolute_import
+
+import os
+import uuid
+
+from contextlib import contextmanager
+from collections import Iterator
+
+try:
+    from urlparse import urlparse
+except ImportError:
+    from urllib.parse import urlparse
+
+import pandas as pd
+from toolz import memoize
+
+import boto
+
+from into import discover, CSV, resource, append, convert, drop, Temp, JSON
+from into import JSONLines, SSH, into, chunks, HDFS
+
+from .text import TextFile
+from ..utils import tmpfile, ext, sample
+
+
+@memoize
+def get_s3_connection(aws_access_key_id=None,
+                      aws_secret_access_key=None,
+                      anon=False):
+    cfg = boto.Config()
+
+    if aws_access_key_id is None:
+        aws_access_key_id = cfg.get('Credentials', 'aws_access_key_id')
+
+    if aws_access_key_id is None:
+        aws_access_key_id = os.environ.get('AWS_ACCESS_KEY_ID')
+
+    if aws_secret_access_key is None:
+        aws_secret_access_key = cfg.get('Credentials', 'aws_secret_access_key')
+
+    if aws_secret_access_key is None:
+        aws_secret_access_key = os.environ.get('AWS_SECRET_ACCESS_KEY')
+
+    # anon is False but we didn't provide any credentials so try anonymously
+    anon = (not anon and
+            aws_access_key_id is None and
+            aws_secret_access_key is None)
+    return boto.connect_s3(aws_access_key_id, aws_secret_access_key,
+                           anon=anon)
```

Comment:
What do you mean by stale in this context? I'm assuming that you won't be accessing things with different credentials on the same machine (note that you *can* do this, I deemed more-than-minimal authentication schemes as low priority while I was working on this). With SSH it matters more because the information to connect is specified in the URI whereas here it isn't.